cli/internal/prune/prune.go

```diff
@@ -146,10 +146,16 @@ func (p *prune) prune(opts *opts) error {
 			continue
 		}
 		workspaces = append(workspaces, ctx.PackageInfos[internalDep].Dir)
+		originalDir := p.base.RepoRoot.UntypedJoin(ctx.PackageInfos[internalDep].Dir.ToStringDuringMigration())
```

Comment:
I think this is supposed to be `ctx.PackageInfos[internalDep].Dir.RestoreAnchor(p.base.RepoRoot)`?

@nathanhammond what do you think about also including a method on `AbsoluteSystemPath` that is essentially `Join` but accepts a single `AnchoredSystemPath` instance?

At any rate, this *shouldn't* be `Untyped`, we have the types here for both pieces of the path.

Reply:
@gsoltis @chris-olszewski I'm game for that method; I just couldn't find a name for it that I was happy with so I didn't create it.

I agree that it isn't intuitive to start with the Anchored path and prepend the Absolute one.